internal/impl/confluent/processor_schema_registry_decode.go

```diff
@@ -323,24 +337,116 @@ func (s *schemaRegistryDecoder) getDecoder(id int) (schemaDecoder, error) {
 	}
 
 	resPayload := struct {
+		Type   string `json:"schemaType"`
 		Schema string `json:"schema"`
 	}{}
 	if err = json.Unmarshal(resBytes, &resPayload); err != nil {
 		s.logger.Errorf("failed to parse response for schema '%v': %v", id, err)
 		return nil, err
 	}
 
-	var codec *goavro.Codec
-	if s.avroRawJSON {
-		if codec, err = goavro.NewCodecForStandardJSONFull(resPayload.Schema); err != nil {
-			s.logger.Errorf("failed to parse response for schema subject '%v': %v", id, err)
-			return nil, err
-		}
+	var (
+		decoder schemaDecoder
+	)
+
+	// https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-protobuf.html#schema-references-in-protobuf
+	// "For backward compatibility reasons, both "schemaType" and "references" are optional. If "schemaType" is omitted, it is assumed to be AVRO."
+	// JSON schema is not supported yet.
+	if resPayload.Type == "PROTOBUF" {
```

Comment:
what are the other types here?

Reply:
`schema_registry_decode` only supports avro for now. There are 3 types in total: https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/index.html#
I can consider contribute json schema in the future as well